### ISSUE TITLE
Maintenance mode after update to Polls 7.1.1 from 7.0.1, <s>and Polls still broken after occ upgrade</s>

### ISSUE THREAD

**Issue by tormodvolden** [nextcloud/polls#3561]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### What went wrong, what did you observe?

This was actually already reported in #3551 but that issue was closed for some reason.

I updated the polls app from 7.0.1 to 7.1.1 and this made Nextcloud go into maintenance mode. I had to run `occ upgrade` to get Nextcloud going again. (This is on Nextcloud 29.0.2)

However, afterwards the Polls app just displays an empty page.

### What did you expect, how polls should behave instead?

I expected the update from 7.0.1 to 7.1.1 to finish without bringing Nextcloud into maintenance mode and without needing to run occ upgrade manually.

I also expected Polls to work correctly after the upgrade.

### What steps does it need to replay this bug?

1. Have Nextcloud 29.0.2 installed with Polls 7.01 working fine
2. Upgrade to Polls 7.1.1 (the only app that needed upgrade)

### Installation method

Installed/updated from the appstore (Apps section of your site)

### Installation type

Updated from a minor version within same major version (i.e. 4.0.0 to 4.1.1)

### Affected polls version

7.1.1

### Which browser did you use, when experiencing the bug?

- [X] Firefox
- [ ] Chrome
- [ ] Chromium/Chromium based (i.e. Edge)
- [ ] Safari
- [ ] Other/Don't know

### NC version

Nextcloud 29

### Other Nextcloud version

Nextcloud 29.0.2

### PHP engine version

PHP 8.2

### Other PHP version

PHP 8.2.20

### Database engine

PostgreSQL

### Database Engine version or other Database

PostgreSQL 16.2

### Which user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other/Don't know

### Add your nextcloud server log here
There is a lot about Repair and dirty tables in the logs, but that was already before the update. I don't know where to start.

This one was about 'related_ressources' and 'Deck' so not interesting:

<details>

```
json
{
  "reqId": "IH2RZ72ntsHzEhPUOoOk",
  "level": 3,
  "time": "2024-06-12T20:01:14+00:00",
  "remoteAddr": "x.x.x.x",
  "user": "oneuser",
  "app": "related_resources",
  "method": "GET",
  "url": "/ocs/v2.php/apps/related_resources/related/account?itemId=anotheruser&resourceType=deck&limit=5&format=json",
  "message": "Exception thrown: OCA\\RelatedResources\\Exceptions\\RelatedResourceProviderNotFound",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; rv:126.0) Gecko/20100101 Firefox/126.0",
  "version": "29.0.2.2",
  "exception": {
    "Exception": "OCA\\RelatedResources\\Exceptions\\RelatedResourceProviderNotFound",
    "Message": "",
    "Code": 0,
    "Trace": [
      {
        "file": "/usr/share/webapps/nextcloud/apps/related_resources/lib/Service/RelatedService.php",
        "line": 173,
        "function": "getRelatedResourceProvider",
        "class": "OCA\\RelatedResources\\Service\\RelatedService",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/apps/related_resources/lib/Service/RelatedService.php",
        "line": 127,
        "function": "retrieveRelatedToItem",
        "class": "OCA\\RelatedResources\\Service\\RelatedService",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/apps/related_resources/lib/Controller/ApiController.php",
        "line": 107,
        "function": "getRelatedToItem",
        "class": "OCA\\RelatedResources\\Service\\RelatedService",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 232,
        "function": "getRelatedResources",
        "class": "OCA\\RelatedResources\\Controller\\ApiController",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 138,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php",
        "line": 184,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/lib/private/Route/Router.php",
        "line": 338,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::"
      },
      {
        "file": "/usr/share/webapps/nextcloud/ocs/v1.php",
        "line": 66,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->"
      },
      {
        "file": "/usr/share/webapps/nextcloud/ocs/v2.php",
        "line": 23,
        "args": ["/usr/share/webapps/nextcloud/ocs/v1.php"],
        "function": "require_once"
      }
    ],
    "File": "/usr/share/webapps/nextcloud/apps/related_resources/lib/Service/RelatedService.php",
    "Line": 605,
    "message": "",
    "exception": {},
    "CustomMessage": "Exception thrown: OCA\\RelatedResources\\Exceptions\\RelatedResourceProviderNotFound"
  }
}

```

</details>

**Comment by dartcafe** [comment]
Your log extract has nothing to do with polls, but with 'related_ressources' and 'Deck'. 
Search for error logs regarding Polls in the log viewer and paste the raw entry here.

**Comment by tormodvolden** [comment]
settings/admin/logging only shows a blank page.

**Comment by tormodvolden** [comment]
This one is maybe more interesting:
`{"reqId":"PrHiY0alMZk5iMPihNMh","level":3,"time":"2024-03-01T08:38:54+00:00","remoteAddr":"x.x.x.x","user":"oneuser","app":"index","method":"GET","url":"/settings/apps/update/polls","message":"Call to undefined method OCA\\Polls\\Db\\VoteMapper::joinAnon() in file '/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php' line 239","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.2.5","exception":{"Exception":"Exception","Message":"Call to undefined method OCA\\Polls\\Db\\VoteMapper::joinAnon() in file '/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php' line 239","Code":0,"Trace":[{"file":"/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/usr/share/webapps/nextcloud/lib/base.php","line":1069,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/usr/share/webapps/nextcloud/index.php","line":39,"function":"handleRequest","class":"OC","type":"::"}],"File":"/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","Line":169,"Previous":{"Exception":"Error","Message":"Call to undefined method OCA\\Polls\\Db\\VoteMapper::joinAnon()","Code":0,"Trace":[{"file":"/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php","line":210,"function":"buildQuery","class":"OCA\\Polls\\Db\\VoteMapper","type":"->"},{"file":"/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php","line":55,"function":"find","class":"OCA\\Polls\\Db\\VoteMapper","type":"->"},{"file":"/var/lib/nextcloud/apps/polls/lib/Db/TableManager.php","line":433,"function":"update","class":"OCA\\Polls\\Db\\VoteMapper","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/lib/nextcloud/apps/polls/lib/Migration/RepairSteps/UpdateHashes.php","line":51,"function":"migrateOptionsToHash","class":"OCA\\Polls\\Db\\TableManager","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OCA\\Polls\\Migration\\RepairSteps\\UpdateHashes","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/legacy/OC_App.php","line":836,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/legacy/OC_App.php","line":777,"function":"executeRepairSteps","class":"OC_App","type":"::"},{"file":"/usr/share/webapps/nextcloud/lib/private/Installer.php","line":202,"function":"updateApp","class":"OC_App","type":"::","args":["*** sensitive parameters replaced ***"]},{"file":"/usr/share/webapps/nextcloud/apps/settings/lib/Controller/AppSettingsController.php","line":538,"function":"updateAppstoreApp","class":"OC\\Installer","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":230,"function":"updateApp","class":"OCA\\Settings\\Controller\\AppSettingsController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":137,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/usr/share/webapps/nextcloud/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/usr/share/webapps/nextcloud/lib/base.php","line":1069,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/usr/share/webapps/nextcloud/index.php","line":39,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php","Line":239},"message":"Call to undefined method OCA\\Polls\\Db\\VoteMapper::joinAnon() in file '/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php' line 239","exception":{},"CustomMessage":"Call to undefined method OCA\\Polls\\Db\\VoteMapper::joinAnon() in file '/var/lib/nextcloud/apps/polls/lib/Db/VoteMapper.php' line 239"}}`

**Comment by tormodvolden** [comment]
Or this:

```
{
    "reqId": "Dt4UG4zAeECNx2fxUKDZ",
    "level": 3,
    "time": "2024-06-12T19:54:49+00:00",
    "remoteAddr": "x.x.x.x",
    "user": "oneuser",
    "app": "index",
    "method": "GET",
    "url": "/settings/apps/update/polls",
    "message": "OCA\\Polls\\Db\\UserMapper::__construct(): Argument #2 ($userManager) must be of type OCP\\IUserManager, OC\\Session\\CryptoSessionData given, called in /var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php on line 158 in file '/var/lib/nextcloud/apps/polls/lib/Db/UserMapper.php' line 43",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; rv:126.0) Gecko/20100101 Firefox/126.0",
    "version": "29.0.2.2",
    "exception": {
        "Exception": "Exception",
        "Message": "OCA\\Polls\\Db\\UserMapper::__construct(): Argument #2 ($userManager) must be of type OCP\\IUserManager, OC\\Session\\CryptoSessionData given, called in /var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php on line 158 in file '/var/lib/nextcloud/apps/polls/lib/Db/UserMapper.php' line 43",
        "Code": 0,
        "Trace": [
            {
                "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php",
                "line": 184,
                "function": "dispatch",
                "class": "OC\\AppFramework\\Http\\Dispatcher",
                "type": "->"
            },
            {
                "file": "/usr/share/webapps/nextcloud/lib/private/Route/Router.php",
                "line": 338,
                "function": "main",
                "class": "OC\\AppFramework\\App",
                "type": "::"
            },
            {
                "file": "/usr/share/webapps/nextcloud/lib/base.php",
                "line": 1050,
                "function": "match",
                "class": "OC\\Route\\Router",
                "type": "->"
            },
            {
                "file": "/usr/share/webapps/nextcloud/index.php",
                "line": 49,
                "function": "handleRequest",
                "class": "OC",
                "type": "::"
            }
        ],
        "File": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "Line": 170,
        "Previous": {
            "Exception": "TypeError",
            "Message": "OCA\\Polls\\Db\\UserMapper::__construct(): Argument #2 ($userManager) must be of type OCP\\IUserManager, OC\\Session\\CryptoSessionData given, called in /var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php on line 158",
            "Code": 0,
            "Trace": [
                {
                    "file": "/var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php",
                    "line": 158,
                    "function": "__construct",
                    "class": "OCA\\Polls\\Db\\UserMapper",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 175,
                    "function": "OCA\\Polls\\AppInfo\\{closure}",
                    "class": "OCA\\Polls\\AppInfo\\Application",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php",
                    "line": 122,
                    "function": "OC\\AppFramework\\Utility\\{closure}",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 142,
                    "function": "offsetGet",
                    "class": "Pimple\\Container",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 464,
                    "function": "query",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 442,
                    "function": "queryNoFallback",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 64,
                    "function": "query",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "file": "/var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php",
                    "line": 198,
                    "function": "get",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 175,
                    "function": "OCA\\Polls\\AppInfo\\{closure}",
                    "class": "OCA\\Polls\\AppInfo\\Application",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php",
                    "line": 122,
                    "function": "OC\\AppFramework\\Utility\\{closure}",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 142,
                    "function": "offsetGet",
                    "class": "Pimple\\Container",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 464,
                    "function": "query",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 442,
                    "function": "queryNoFallback",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 96,
                    "function": "query",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "function": "OC\\AppFramework\\Utility\\{closure}",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 83,
                    "function": "array_map"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 128,
                    "function": "buildClass",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 146,
                    "function": "resolve",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 470,
                    "function": "query",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 442,
                    "function": "queryNoFallback",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 96,
                    "function": "query",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "function": "OC\\AppFramework\\Utility\\{closure}",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 83,
                    "function": "array_map"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 128,
                    "function": "buildClass",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 146,
                    "function": "resolve",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php",
                    "line": 470,
                    "function": "query",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/ServerContainer.php",
                    "line": 155,
                    "function": "queryNoFallback",
                    "class": "OC\\AppFramework\\DependencyInjection\\DIContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php",
                    "line": 64,
                    "function": "query",
                    "class": "OC\\ServerContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/Repair.php",
                    "line": 141,
                    "function": "get",
                    "class": "OC\\AppFramework\\Utility\\SimpleContainer",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/legacy/OC_App.php",
                    "line": 831,
                    "function": "addStep",
                    "class": "OC\\Repair",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/legacy/OC_App.php",
                    "line": 774,
                    "function": "executeRepairSteps",
                    "class": "OC_App",
                    "type": "::"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/Installer.php",
                    "line": 184,
                    "function": "updateApp",
                    "class": "OC_App",
                    "type": "::",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/apps/settings/lib/Controller/AppSettingsController.php",
                    "line": 605,
                    "function": "updateAppstoreApp",
                    "class": "OC\\Installer",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
                    "line": 232,
                    "function": "updateApp",
                    "class": "OCA\\Settings\\Controller\\AppSettingsController",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
                    "line": 138,
                    "function": "executeController",
                    "class": "OC\\AppFramework\\Http\\Dispatcher",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php",
                    "line": 184,
                    "function": "dispatch",
                    "class": "OC\\AppFramework\\Http\\Dispatcher",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/private/Route/Router.php",
                    "line": 338,
                    "function": "main",
                    "class": "OC\\AppFramework\\App",
                    "type": "::"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/lib/base.php",
                    "line": 1050,
                    "function": "match",
                    "class": "OC\\Route\\Router",
                    "type": "->"
                },
                {
                    "file": "/usr/share/webapps/nextcloud/index.php",
                    "line": 49,
                    "function": "handleRequest",
                    "class": "OC",
                    "type": "::"
                }
            ],
            "File": "/var/lib/nextcloud/apps/polls/lib/Db/UserMapper.php",
            "Line": 43
        },
        "message": "OCA\\Polls\\Db\\UserMapper::__construct(): Argument #2 ($userManager) must be of type OCP\\IUserManager, OC\\Session\\CryptoSessionData given, called in /var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php on line 158 in file '/var/lib/nextcloud/apps/polls/lib/Db/UserMapper.php' line 43",
        "exception": {

        },
        "CustomMessage": "OCA\\Polls\\Db\\UserMapper::__construct(): Argument #2 ($userManager) must be of type OCP\\IUserManager, OC\\Session\\CryptoSessionData given, called in /var/lib/nextcloud/apps/polls/lib/AppInfo/Application.php on line 158 in file '/var/lib/nextcloud/apps/polls/lib/Db/UserMapper.php' line 43"
    }
}

```

**Comment by q-wertz** [comment]
Not sure if it helps with the update to polls 7.1.1 but what I did after the broken update to 7.1.0 (on 3 instances) was 
```
sudo -u www-data php occ update
sudo -u www-data php occ maintenance:mode --off
```
and Polls 7.1.0 was working afterwards (as far as I can tell).

The update from 7.1.0 to 7.1.1 worked without any issues (for me)

**Comment by tormodvolden** [comment]
@q-wertz Maybe you meant `upgrade` instead of `update`?

Anyway I had to run this in the first place to get Nextcloud online again. But the Polls app remains broken.

**Comment by q-wertz** [comment]
@tormodvolden you are right

Already deleted the comment because I didn't read your issue well enough. Didn't see you also performed this step, thought you only disabled the maintenance mode.

As it is deleted:
Can only tell you that by performing the "broken" update to 7.1.0 first (required `occ upgrade` and `occ maintenance:mode --off`) and now doing the update to 7.1.1 (without issues) the polls app seem to work. Maybe this also helps the devs.

**Comment by dartcafe** [comment]
@tormodvolden Repeating the app update should fix this. 

The error is caused because new methods are not registered in the update process, when they are executed/requested the first time. See https://github.com/nextcloud/server/issues/38158

**Comment by tormodvolden** [comment]
Great, but how do I repeat the update?
```
$ occ app:update polls
polls is up-to-date or no updates could be found

```

**Comment by tormodvolden** [comment]
I edited the apps/polls/appinfo/info.xml file to say "7.1.0" in order to trick nextcloud to "update" to 7.1.1 again.
```
$ occ app:update polls
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
polls new version available: 7.1.1
dlopen: libavcodec.so.60: cannot open shared object file: No such file or directory
polls updated
$ 
$ occ app:update polls
polls is up-to-date or no updates could be found
$ 

```
(The libavcodec error has been there for a long time, doesn't seem to matter)

But Polls still only shows an empty page.

**Comment by q-wertz** [comment]
I think you need to perform

```
sudo -u www-data ./occ upgrade
```
not `occ app:update polls`
in your current state

**Comment by tormodvolden** [comment]
```
$ occ upgrade         
Nextcloud is already latest version
$ 
```

**Comment by q-wertz** [comment]
:disappointed: 

If I remember correctly I had quite some output after the failed polls update.

**Comment by dartcafe** [comment]
I don't know what is happening at your system. What exactly is now the error description?
First report was 'Maintenance mode' and a broken install. Now the app seems to be installed, but the page stays empty. 

Maybe you should remove the app and the reinstall and refresh your browser cache.

**Comment by tormodvolden** [comment]
Nothing has changed. In my original report it was first "Maintenance mode" and after "occ upgrade" it was a broken Polls (empty page), although the app was reportedly installed. This is still the case, after trying to reinstall it.

**Comment by tormodvolden** [comment]
OK, so I disabled and removed the app in the GUI, then reinstalled it:
```
$ occ app:install polls
dlopen: libavcodec.so.60: cannot open shared object file: No such file or directory
polls 7.1.1 installed
polls enabled
$
```
 I also restarted the php-fpm daemon. Still no dice, the page is empty.

**Comment by dartcafe** [comment]
Any more error  logging inside the server log or the browser?

**Comment by tormodvolden** [comment]
There are no new entries in the nextcloud.log since the update on the 12th (except one entry from /settings/user/sync-clients coming from me perusing all user settings in the GUI yesterday).

In the web server log there are some 404's on URLs starting with /wapps/ concerning several apps. After the update from 7.0.1 there is this showing up:
`"GET /wapps/polls/js/polls-main.mjs?v=xxxxxx-x HTTP/1.1" 404`
Unfortunately I am not sure we have logs going long enough back to be able to tell whether /wapps/polls was requested with the 7.0.1 version.

This is also reflected in the browser console:
`Loading failed for the module with source “https://cloud.laborluzern.ch/wapps/polls/js/polls-main.mjs?v=xxxxxx-x”. NcButton-DhaPcomf.css:32:20
`

The other 404's (not mentioning Polls) are seen here as:
`Source map error: Error: request failed with status 404
`

I had once an issue with "wapps" and it was a bug in an app: https://github.com/pierre-alain-b/rainloop-nextcloud/issues/274

**Comment by tormodvolden** [comment]
There are no new entries in the nextcloud.log since the update on the 12th (except one entry from /settings/user/sync-clients coming from me perusing all user settings in the GUI yesterday).

In the web server log there are some 404's on URLs starting with /wapps/ concerning several apps. After the update from 7.0.1 there is this showing up:
`"GET /wapps/polls/js/polls-main.mjs?v=xxxxxx-x HTTP/1.1" 404`
Unfortunately I am not sure we have logs going long enough back to be able to tell whether /wapps/polls was requested with the 7.0.1 version.

This is also reflected in the browser console:
`Loading failed for the module with source “https://server.domain/wapps/polls/js/polls-main.mjs?v=xxxxxx-x”. NcButton-DhaPcomf.css:32:20
`

The other 404's (not mentioning Polls) are seen here as:
`Source map error: Error: request failed with status 404
`

I had once an issue with "wapps" and it was a bug in an app: https://github.com/pierre-alain-b/rainloop-nextcloud/issues/274

**Comment by tormodvolden** [comment]
A complete dump from the browser console. "disallowed MIME type" is interesting.

<details>

```
Content-Security-Policy: Ignoring “blob:” within script-src-elem: ‘strict-dynamic’ specified polls
GET
https://server.domain/wapps/polls/js/polls-main.mjs?v=xxxxxxxx-x

GET
https://server.domain/apps/firstrunwizard/js/firstrunwizard-activate.mjs?v=xxxxxxxx-x

GET
https://server.domain/apps/firstrunwizard/js/firstrunwizard-about.mjs?v=xxxxxxxx-x

Loading module from “https://server.domain/wapps/polls/js/polls-main.mjs?v=xxxxxxxx-x” was blocked because of a disallowed MIME type (“text/html”). polls
Loading failed for the module with source “https://server.domain/wapps/polls/js/polls-main.mjs?v=xxxxxxxx-x”. polls:32:208
Loading module from “https://server.domain/apps/firstrunwizard/js/firstrunwizard-activate.mjs?v=xxxxxxxx-x” was blocked because of a disallowed MIME type (“text/html”). polls
Loading failed for the module with source “https://server.domain/apps/firstrunwizard/js/firstrunwizard-activate.mjs?v=xxxxxxxx-x”. polls:37:229
Loading module from “https://server.domain/apps/firstrunwizard/js/firstrunwizard-about.mjs?v=xxxxxxxx-x” was blocked because of a disallowed MIME type (“text/html”). polls
Loading failed for the module with source “https://server.domain/apps/firstrunwizard/js/firstrunwizard-about.mjs?v=xxxxxxxx-x”. polls:38:226
GET
https://server.domain/apps/theming/image/logoheader?v=9
[HTTP/1.1 404 Not Found 478ms]

Unified search initialized! core-unified-search.js:2:28293
Initializing unified search plugin-filters from talk talk-search.js:782:65
session heartbeat polling started core-main.js:2:29778
GET
https://server.domain/apps/theming/image/logoheader?v=9
[HTTP/1.1 404 Not Found 383ms]

Registering notifications container as a menu notifications-src_NotificationsApp_vue.js:2:17954
Notifications permissions denied notifications-src_NotificationsApp_vue.js:2:23104
Polling interval updated to 30000 notifications-src_NotificationsApp_vue.js:2:22500
Started background fetcher as session_keepalive is enabled notifications-src_NotificationsApp_vue.js:2:19291
Source map error: Error: request failed with status 404
Resource URL: https://server.domain/dist/core-unified-search.js?v=xxxxxxxx-x
Source Map URL: core-unified-search.js.map?v=xxxxxxxxxxxxxxxxxxxx
Source map error: Error: request failed with status 404
Resource URL: https://server.domain/dist/core-main.js?v=xxxxxxxx-x
Source Map URL: core-main.js.map?v=xxxxxxxxxxxxxxxxxxxx
Source map error: Error: request failed with status 404
Resource URL: https://server.domain/wapps/spreed/js/talk-search.js?v=xxxxxxxx-x
Source Map URL: talk-search.js.map?v=xxxxxxxxxxxxxxxxxxxx
Search providers 
Array(15) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
core-unified-search.js:2:15546
Contacts 
Array(25) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
core-unified-search.js:2:15657
Got notification data, restoring default polling interval. notifications-src_NotificationsApp_vue.js:2:21580
Source map error: Error: request failed with status 404
Resource URL: https://server.domain/apps/notifications/js/notifications-src_NotificationsApp_vue.js?v=xxxxxxxxxxxxxxxxxxxx
Source Map URL: notifications-src_NotificationsApp_vue.js.map?v=xxxxxxxxxxxxxxxxxxxx

```
</details>

**Comment by tormodvolden** [comment]
I noticed that it was often .mjs files that appeared in the errors. I checked which apps are using .mjs files and there are not too many. All these apps had issues, it turned out. I had to edit my /etc/nginx/conf/nextcloud.conf and add "mjs" to the list here:
```
    # Adding the cache control header for js and css files
    # Make sure it is BELOW the PHP block
    location ~ \.(?:css|mjs|js|woff2?|svg|gif)$ {
```
Now Polls and the other apps (activity, logging) work.

**Comment by steviehs** [comment]
On my side it worked only after I added 
`application/javascript                mjs;` 
to
`/etc/nginx/mime.types`

**Comment by q-wertz** [comment]
I think those two settings are also described in the official [admin manual](https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html)

**Comment by rasos** [comment]
During the app update to Polls 7.1.1. we had a memory exhaust: 

>   [Tue Nov 22 12:14:23.035667 2022] [proxy_fcgi:error] [pid 3536834:tid 140701828368128] [client 83.65.86.98:62539] AH01071: Got error 'PHP message: PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 28672 bytes) in /var/www/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php on line 999'

nextcloud went into maintenance mode and the only way out was doing a nextcloud update. 

With NC 27.1.10 and enabling again the Polls app 7.1.1 everything is fine and we can see the polls.

**Comment by kinimodmeyer** [comment]
I fear i had the same issue, now when i click on a list to view the polls i get MySQL errors in the nextcloud file "Did expect one result but found none when executing".

```
query \"SELECT `polls_polls`.*, (SELECT COUNT(`user_vote_sub`.`vote_answer`) FROM `*PREFIX*polls_votes` `user_vote_sub` WHERE (`user_vote_sub`.`poll_id` = `polls_polls`.`id`) AND (`user_vote_sub`.`user_id` = :dcValue1)) AS `current_user_count_votes`, (SELECT COUNT(`user_vote_sub`.`vote_answer`) FROM `*PREFIX*polls_votes` `user_vote_sub` WHERE (`user_vote_sub`.`poll_id` = `polls_polls`.`id`) AND (`user_vote_sub`.`user_id` = :dcValue1) AND (`user_vote_sub`.`vote_answer` = :dcValue2)) AS `current_user_count_votes_yes`, (SELECT COUNT(`user_vote_sub`.`vote_answer`) FROM `*PREFIX*polls_votes` `user_vote_sub` LEFT JOIN `*PREFIX*polls_options` `vote_options_join` ON (`vote_options_join`.`poll_id` = `user_vote_sub`.`poll_id`) AND (`vote_options_join`.`poll_option_text` = `user_vote_sub`.`vote_option_text`) AND (`vote_options_join`.`deleted` = '0') WHERE (`user_vote_sub`.`poll_id` = `polls_polls`.`id`) AND (`user_vote_sub`.`user_id` = :dcValue1) AND (`vote_options_join`.`id` IS NULL)) AS `current_user_count_orphaned_votes`, coalesce(MAX(options.timestamp), :dcValue3) AS max_date, coalesce(MIN(options.timestamp), :dcValue4) AS min_date, COUNT(`options`.`id`) AS `count_options`, coalesce(user_shares.type, '') AS user_role, `user_shares`.`locked` AS `is_current_user_locked`, coalesce(user_shares.token, '') AS share_token, group_concat(distinct group_shares.user_id SEPARATOR \",\") AS group_shares FROM `*PREFIX*polls_polls` `polls_polls` LEFT JOIN `*PREFIX*polls_options` `options` ON (`polls_polls`.`id` = `options`.`poll_id`) AND (`options`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `user_shares` ON (`polls_polls`.`id` = `user_shares`.`poll_id`) AND (`user_shares`.`user_id` = :dcValue5) AND (`user_shares`.`deleted` = '0') LEFT JOIN `*PREFIX*polls_share` `group_shares` ON (`polls_polls`.`id` = `group_shares`.`poll_id`) AND (`group_shares`.`type` = 'group') AND (`group_shares`.`deleted` = '0') WHERE `polls_polls`.`id` = :dcValue6 GROUP BY `polls_polls`.`id`, `user_shares`.`type`, `user_shares`.`locked`, `user_shares`.`token`\"; 
```

**Comment by bwhb** [comment]
I have the same problem. Empty poll page after updating to 7.1.1 on 29.0.2.
Do I lose the data if I reinstall the app?

**Comment by dartcafe** [comment]
> I think those two settings are also described in the official [admin manual](https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html)

Thanks for caring and support. Learned something. 👍

**Comment by dartcafe** [comment]
To all other guys. I see different issues here in this thread. It is not possible to track these inside one issue. 

Please be so kind and open a [new issue ](https://github.com/nextcloud/polls/issues/new?assignees=&labels=bug&projects=&template=BUG_REPORT.yml) and answer the questions, so I am able to analyse your issues.
